Makefile

```diff
@@ -79,6 +73,15 @@ tarball: npm_licenses common-tarball
 .PHONY: docker
 docker: npm_licenses common-docker
 
+.PHONY: common-build
+common-build: promu
+	@bash -c "trap 'echo \">> restoring assets\"; scripts/compress_assets.sh -d;' EXIT; \
+	          echo '>> compressing assets'; \
+	          scripts/compress_assets.sh; \
```

Comment:
Why not having this in "assets" ?

Reply:
Apologies if I misunderstand...
This compresses the assets that are specific to prometheus and are located in this repo. For, say alertmanager the assets are different and in a different location. I'm not sure how that could be handled from the common repo.

Reply:
I mean, put this under "make assets" not "make build"

Reply:
Ahh I see.
The main idea is that I can clean up (i.e. decompress) the assets in case the build fails. If we put the compression into the asset target, we need an extra `decompress` target. And it also becomes more awkward to run this automatically in case the build fails.
I'd even argue this is the right place conceptually. The assets are ready to go as is, the compression step is really an optimization to keep the binary size a bit smaller. Building prometheus without the compression step still yields a correct binary, its just ~10 MB larger.

Reply:
I still think it should be part of make assets.

Reply:
I also think that using trap is an advanced move, I am not sure we should do this.